Bottom: 1518c4bad10116380850a0ffdac30ff7ae154631
Top:    c6a59c7ede01ea94c086c8907166c8e68869e4cf
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-09-05 11:57:27 +0100

hid/gtk: Avoid double-destroying the file-changed info_bar on "Reload"

This was caused by an addition I made in the earlier patch to destroy
the info_bar when the user manually reverts or loads a new file).

The callback on revert button was trying to delete it (again) after
the revert _action_ we call triggered a destroy of the widget. Avoid
this by destroying the info_bar before we call the revert action.


---

diff --git a/src/hid/gtk/gui-top-window.c b/src/hid/gtk/gui-top-window.c
index f713446..f3e95e7 100644
--- a/src/hid/gtk/gui-top-window.c
+++ b/src/hid/gtk/gui-top-window.c
@@ -471,11 +471,11 @@ info_bar_response_cb (GtkInfoBar *info_bar,
 {
   GhidGui *_gui = (GhidGui *)user_data;
 
-  if (response_id == GTK_RESPONSE_ACCEPT)
-    hid_actionl ("LoadFrom", "revert", "none", NULL);
-
   gtk_widget_destroy (_gui->info_bar);
   _gui->info_bar = NULL;
+
+  if (response_id == GTK_RESPONSE_ACCEPT)
+    hid_actionl ("LoadFrom", "revert", "none", NULL);
 }
 
 static void
